main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala

```diff
@@ -732,15 +732,18 @@ object GenExpression {
         compileExpr(exp)
         ARRAYLENGTH()
 
-      case AtomicOp.StructNew(sym, _) =>
+      case AtomicOp.StructNew(sym, mutability, _) =>
         import BytecodeInstructions.*
-
-        val region :: fieldExps = exps
         val structType = getStructType(root.structs(sym))
-
-        // Evaluate the region and ignore its value
-        compileExpr(region)
-        xPop(BackendType.toBackendType(region.tpe))
+        val fieldExps = mutability match {
```

Comment:
I slightly dislike that you are both: functionally computing a list of fields AND conditionally emitting bytecode for one expression. Maybe its better to duplicate the 4 lines after and just have each case be complete in its own.

Reply:
(This is not backtracking on what I said earlier: My point was only to have one match on StructNew. The inner exp can then match on mutability and then there can be to separate code paths.)